pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -67,27 +65,15 @@ func IsReadyToReconcile(c client.Client, clustercontext *clusterd.Context, names
 
 	logger.Debugf("%q: CephCluster resource %q found in namespace %q", controllerName, cephCluster.Name, namespacedName.Namespace)
 
-	// If the cluster is healthy
-	// Test a Ceph command to verify the Operator is ready
-	// This is done to silence errors when the operator just started and cannot reconcile yet
-	status, err := cephclient.Status(clustercontext, namespacedName.Namespace)
-	if err != nil {
-		if strings.Contains(err.Error(), "error calling conf_read_file") {
-			logger.Infof("%q: operator is not ready to run ceph command, cannot reconcile yet.", controllerName)
-			return cephCluster, false, cephClusterExists, WaitForRequeueIfCephClusterNotReady
+	// read the CR status of the cluster
+	if cephCluster.Status.CephStatus != nil {
+		if cephCluster.Status.CephStatus.Health == "HEALTH_OK" || cephCluster.Status.CephStatus.Health == "HEALTH_WARN" {
+			logger.Debugf("%q: ceph status is %q, operator is ready to run ceph command, reconciling", controllerName, cephCluster.Status.CephStatus.Health)
+			return cephCluster, true, cephClusterExists, WaitForRequeueIfCephClusterNotReady
 		}
-		// We should not arrive there
-		logger.Errorf("%q: ceph command error %v", controllerName, err)
-		return cephCluster, false, cephClusterExists, ImmediateRetryResult
-	}
-
-	// If Ceph status is ok we can reconcile
-	if status.Health.Status == "HEALTH_OK" || status.Health.Status == "HEALTH_WARN" {
-		logger.Debugf("%q: ceph status is %q, operator is ready to run ceph command, reconciling", controllerName, status.Health.Status)
-		return cephCluster, true, cephClusterExists, reconcile.Result{}
+		logger.Infof("%s: CephCluster %q found but skipping reconcile since ceph health is %q", controllerName, cephCluster.Name, cephCluster.Status.CephStatus)
```

Comment:
This is for a separate discussion, but I still don't see why we block reconciling even if the ceph health has an error. For example, if mds raises a health error, it shouldn't block the reconcile of object stores or object users.

Reply:
That's a good point actually, I'm fine letting the controller reconcile though. It's probably what we should do anyway. Follow-up PR in master?